Jason dev

biodataganache opened the pull request:
Updating the min_filter to allow for percentages as input. Also add a helpful error message that is printed and added to the log file if the user has filtered out all the kmers using min_filter.

biodataganache committed:
Adding support for min_filter as a percentage of kmers

biodataganache committed:
Added min_filter error message and logfile output

biodataganache committed:
Minor change

christinehc commented:
Hmm, not sure if use is as intended, but I'm getting the same full_feature_matrix shape when `min_filter=0.1` and `min_filter=0.9`:

![image](https://user-images.githubusercontent.com/46230556/194956650-469e1b22-54f0-4230-8828-aa04212c3fe6.png)

![image](https://user-images.githubusercontent.com/46230556/194956668-56ddcf78-35aa-4568-812b-5c58ef77ea0b.png)

vs. `min_filter=10`, where it seems kmers have been pared down:

![image](https://user-images.githubusercontent.com/46230556/194956806-4918ba2c-fe8f-47f8-8887-9af8e5d5074e.png)

biodataganache commented:
> Hmm, not sure if use is as intended, but I'm getting the same full_feature_matrix shape when `min_filter=0.1` and `min_filter=0.9`:
> 
> ![image](https://user-images.githubusercontent.com/46230556/194956650-469e1b22-54f0-4230-8828-aa04212c3fe6.png)
> 
> ![image](https://user-images.githubusercontent.com/46230556/194956668-56ddcf78-35aa-4568-812b-5c58ef77ea0b.png)
> 
> vs. `min_filter=10`, where it seems kmers have been pared down:
> 
> ![image](https://user-images.githubusercontent.com/46230556/194956806-4918ba2c-fe8f-47f8-8887-9af8e5d5074e.png)

That's weird. Is this for model?

christinehc commented:
This is for cluster mode

biodataganache commented:
0.999  = (289, 61)
0.1 = (289, 0)
1 = (289, 15796)
For cluster in my hands? Weird.

christinehc commented:
Ah, I realized the issue. I was changing `config['cluster']['min_rep']` and not `config['min_filter']`.

I do think it may be a bit confusing to have both of these parameters named as-is, though I realize they accomplish slightly different things

christinehc committed:
Terminate workflow via error instead of printing to console

Changes the behavior of the `min_filter` parameter when the filter is set too high; previously, an error message was printed to console, but the snakemake workflow will continue running until the missing input files are detected in the next rule. This has been changed to terminate with an error immediately instead. Thus, the workflow will not attempt to run subsequent rules that are invalid past the `min_filter` step.

christinehc commented:
Made changes to raise `ValueError` to terminate the workflow in case not enough kmers pass the filter.

I do still think the `min_filter` vs. `min_rep` may warrant further discussion. I also find it a bit confusing to use the `min_filter` parameter since it is currently a "hidden parameter" and we don't have good documentation for it.

Also, with the current behavior, I think something like `include` would be a better name for the parameter, referring to the fact that `(1 - min_filter)%` of kmers are captured. Otherwise, `filter` seems to imply the reverse to me. Thoughts?

biodataganache commented:
OK - I've changed filtering so it now uses a single global parameter "kmer_include_filter" - but has backwards compatibility with "min_filter" since I realized I just set that in stone in the manuscript (whoops). The filtering for model and cluster are all done in the kmerize rule and I removed the (partially) redundant filtering that was specific to cluster. So no more cluster/min_rep and I removed cluster/max_rep (but we should probably replace it in the future).
Finally - I commented the config.yaml a bunch to be more explanatory about what the parameters mean.

christinehc commented:
I'm not seeing any changes to the `jason-dev` branch-- have those been pushed?

biodataganache commented:
Weird. I don't know what happened to that change :) - will find it.

biodataganache commented:
Ancient PR that's not worth resurrecting now.
